modules/admin-ui-picard/app/src/thunks/tableThunks.js

```diff
@@ -44,308 +44,422 @@ import {fetchThemes} from "./themeThunks";
+    let tableData = {
         resource: "servers",
         rows: resource,
-        columns: columns,
-        multiSelect: multiSelect,
+        columns: table.columns,
+        multiSelect: table.multiSelect,
         pages: pages,
-        sortBy: "online"
+        sortBy: table.sortBy,
+        totalItems: total
     };
+
+    if (table.resource !== 'servers') {
+        const c = serversTableConfig.columns;
+        const columns = c.map(column => {
+            const col = servers.columns.find(co => co.name === column.name);
+            return {
+                ...column,
+                deactivated: col.deactivated
+            }
+        });
+
+        const multiSelect = serversTableConfig.multiSelect;
+
+        tableData = {
+            ...tableData,
+            columns: columns,
+            sortBy: "online",
+            multiSelect: multiSelect
+        };
+    }
     dispatch(loadResourceIntoTable(tableData));
 
 }
 
 export const loadServicesIntoTable = () => (dispatch, getState) => {
-    const { services } = getState();
-    const pagination = getTablePagination(getState());
+    const { services, table } = getState();
+    const pagination = table.pagination;
     const resource = services.results;
+    const total = services.total;
 
-    const c = servicesTableConfig.columns;
-    const columns = c.map(column => {
-        const col = services.columns.find(co => co.name === column.name);
-        return {
-            ...column,
-            deactivated: col.deactivated
-        }
-    });
-
-    const multiSelect = servicesTableConfig.multiSelect;
+    const pages = calculatePages(total / pagination.limit, pagination.offset);
 
-    const pages = calculatePages(resource.length / pagination.limit, pagination.offset);
-
-    const tableData = {
-        resource: "services",
+    let tableData = {
         rows: resource,
-        columns: columns,
-        multiSelect: multiSelect,
         pages: pages,
-        sortBy: "status"
-    };
+        totalItems: total,
+        resource: "services",
+        columns: table.columns,
+        multiSelect: table.multiSelect,
+        sortBy: table.sortBy
+    }
+
+    if (table.resource !== 'services') {
+        const c = servicesTableConfig.columns;
+        const columns = c.map(column => {
+            const col = services.columns.find(co => co.name === column.name);
+            return {
+                ...column,
+                deactivated: col.deactivated
+            }
+        });
+
+        const multiSelect = servicesTableConfig.multiSelect;
+
+        tableData = {
+            ...tableData,
+            columns: columns,
+            sortBy: "status",
+            multiSelect: multiSelect
+        };
+    }
+
     dispatch(loadResourceIntoTable(tableData));
 }
 
 export const loadUsersIntoTable = () => (dispatch, getState) => {
-    const {users} = getState();
+    const { users, table} = getState();
     const pagination = getTablePagination(getState());
```

Comment:
For all other resources, you changed from `getTablePagination(getState())` to `table.pagination`. Is this case different?